driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java

```diff
@@ -238,6 +249,10 @@ private OIndexManager getIndexManager() {
 
     @Override
     public Iterator<Edge> edges(Object... edgeIds) {
+        Set<Class<?>> edgeIdTypes = Stream.of(edgeIds).map(edgeId -> edgeId.getClass()).collect(toSet());
+        if(edgeIdTypes.size() > 1 && edgeIdTypes.contains(OrientEdge.class))
+            throw Graph.Exceptions.idArgsMustBeEitherIdOrElement();
```

Comment:
I find that rather odd. And I guess you do too, as you asked for my thoughts. 
I much prefer positive tests (A does X) over negative tests (A does not do X), as you can spend the rest of your life on the negative tests and only end up with messy code. And in this case, it's also iterating through the edges prematurely, just for that test. 
Can we selectively disable that test?